docs/architecture/dapr-for-net-developers/actors.md

```diff
@@ -0,0 +1,636 @@
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+  - name: actorStateStore
+    value: "true"
+```
+
+## Use the Dapr .NET SDK
+
+You can create an actor model implementation using only HTTP/gRPC calls. However, it's much more convenient to use the language specific Dapr SDKs. At the time of writing, both the .NET and Java SDKs provide extensive support for working with actors.
+
+To get started with the .NET Dapr actors SDK, you add a package reference to [`Dapr.Actors`](https://www.nuget.org/packages/Dapr.Actors) to your service project. The first step of creating an actual actor is to define an interface that derives from `IActor`. Clients use the interface to invoke operations on the actor. Here's a simple example of an actor interface for keeping scores:
+
+```csharp
+public interface IScoreActor : IActor
+{       
+    Task<int> IncrementScoreAsync();
+
+    Task<int> GetScoreAsync();
+}
+```
+
+> [!IMPORTANT]
+> The return type of an actor method must be `Task` or `Task<T>`. Also, actor methods can have at most one argument. Both the return type and the arguments must be `System.Text.Json` serializable.
+
+Next, implement the actor by deriving a `ScoreActor` class from `Actor`. The `ScoreActor` class must also implement the `IScoreActor` interface:
+
+```csharp
+public class ScoreActor : Actor, IScoreActor
+{
+    public ScoreActor(ActorHost host) : base(host)
+    {
+    }
+
+    // TODO Implement interface methods.
+}
+```
+
+The constructor in the snippet above takes a `host` argument of type `ActorHost`. The `ActorHost` class represents the host for an actor type within the actor runtime. You need to pass this argument to the constructor of the `Actor` base class. Actors also support dependency injection. Any additional arguments that you add to the actor constructor are resolved using the .NET dependency injection container.
+
+Let's now implement the `IncrementScoreAsync` method of the interface:
+
+```csharp
+public Task<int> IncrementScoreAsync()
+{
+    return StateManager.AddOrUpdateStateAsync(
+        "score",
+        1,
+        (key, currentScore) => currentScore + 1
+    );
+}
+```
+
+In the snippet above, a single call to `StateManager.AddOrUpdateStateAsync` provides the full implementation for the `IncrementScoreAsync` method. The `AddOrUpdateStateAsync` method takes three arguments:
+
+1. The key of the state to update.
+1. The value to write if no score is stored in the state store yet.
+1. A `Func` to call if there already is a score stored in the state store. It takes the state key and current score, and returns the updated score to write back to the state store.
+
+The `GetScoreAsync` implementation reads the current score from the state store and returns it to the client:
+
+```csharp
+public async Task<int> GetScoreAsync()
+{
+    var scoreValue = await StateManager.TryGetStateAsync<int>("score");
+    if (scoreValue.HasValue)
+    {
+        return scoreValue.Value;
+    }
+
+    return 0;
```

Comment:
If this is a `Nullable<int>` (`int?`), maybe simplify the return to this?

```suggestion
    return scoreValue.GetValueOrDefault();
```

Reply:
`GetStateAsync` returns an instance of `Dapr.Actors.Runtime.ConditionalValue<TValue>` which only has `HasValue()` and `Value` members